### ISSUE TITLE
ShapeList of Compound displayed incorrectly

### ISSUE THREAD

**Issue by MatthiasJ1** [bernhard-42/vscode-ocp-cad-viewer#91]
```py
x = Compound(children=[Circle(1), Pos(3)*Circle(1)])
x = ShapeList([x, Pos(Y=3)*x])

show_all()
```

**Incorrect:**
<img width="618" alt="Screenshot 1" src="https://github.com/user-attachments/assets/07244a38-c18f-458a-a5dc-5140064aeb02">

```py
x, y = x # Expand list x into its components within separate variables

show_all()
```
**Correct:**
<img width="701" alt="Screenshot 2" src="https://github.com/user-attachments/assets/b84bfd69-50bf-4403-9687-ff12138910bc">

**Comment by bernhard-42** [comment]
Doesn't work in the latest version yet.
I never expected someone using `ShapeList` as a wrapper to generic lists.
```python
x = Compound(children=[Circle(1), Pos(3)*Circle(1)])
x = [x, Pos(Y=3)*x]

show_all()
```
works in my latest version.

Out of curiosity, why are you doing this?

**Comment by MatthiasJ1** [comment]
Well in my scenario I was projecting a sketch onto a solid which would give >1 Compounds which I was returning as a ShapeList to make it easy to sort/operate on. In this case Compounds made the most sense to me categorically but in general using unfused compounds can yield significant performance improvements.

**Comment by bernhard-42** [comment]
I now differentiate between (due to internal reasons) homogenous `ShapeList`s (as returned by e.g. `edges()`) and other `ShapeList`s. The other ones will simply be treated as lists and unrolled by the converter.

**Comment by bernhard-42** [comment]
fixed in 2.4.0
